### PR TITLE
move zos client under j.clients.zos instead of the protocol client

### PR DESCRIPTION
**Pull request by abom**
Resolves: #292.

## Description

Using the high level client of zos.

### Before asking for approval

- [ ] Make sure the changes are covered by tests.
- [ ] All the CI tasks are green.
- [ ] Ensure the branch is up to date with its base branch.
- [ ] Corresponding changes to the documentation were made.

### CODE REVIEW

File: Jumpscale/clients/zero_os_protocol/ZeroOSFactory.py [hunk 1]
```diff
@@ -7,7 +7,7 @@ class ZeroOSFactory(j.application.JSBaseConfigsClass):
     """
     """
     _CHILDCLASS = Client
-    __jslocation__ = "j.clients.zos"
+    __jslocation__ = "j.clients.zos_protocol"
```
Comment (zaibon) [inline Jumpscale/clients/zero_os_protocol/ZeroOSFactory.py:10]: This file also needs to be clean up from method like
- get_by_id
- zero_node_ovh_install
- zero_node_packetnet_install
- get_from_itsyouonline
And everything with is not related to the protocol client.

File: Jumpscale/clients/zero_os/ZeroOSClient.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+from Jumpscale import j
+from Jumpscale.sal_zos.node.Node import Node
+
+
+class ZeroOSClient(j.application.JSBaseConfigClass, Node):
+
+    _SCHEMATEXT = """
```
Comment (zaibon) [inline Jumpscale/clients/zero_os/ZeroOSClient.py:7]: I wonder if this is the right place for this schema.
Seems this schema define the config for the protocol client.
For example I tried to create a node like this:
```python
node = j.clients.zos.get('ac1f6b457338',host='10.102.71.171') 
```
Then I realized I also needed to add a password. So I did
```python
node.data.password = '.....'
```
but still I get authentication error when trying to talk to the node. Seems the reason is the protocol client didn't receive the password update in its schema. 
Which leave me with the hack to do
```python
node.client.data.password= '...'
```
Which I believe is wrong. use shouln't have to know about these implementation detail to use the client.

**Review by zaibon — CHANGES_REQUESTED** [review]
Added comments

File: Jumpscale/clients/zero_os/ZeroOSClient.py (hunk 2 above)
Comment (abom) [reply Jumpscale/clients/zero_os/ZeroOSClient.py:7]: It's the same schema defined by zos protocol which is used under the hood, how else would we pass this config? and from where?

the data is passed correctly to the protocol client, can you try with j.clients.zos_protocol only?

e.g.  `j.clients.zos_protocol.get('ac1f6b457338',host='10.102.71.171')`

Comment (zaibon) [reply Jumpscale/clients/zero_os/ZeroOSClient.py:7]: > It's the same schema defined by zos protocol which is used under the hood, how else would we pass this config? and from where?

Well that's the question I'm also asking.

> j.clients.zos_protocol.get('ac1f6b457338',host='10.102.71.171')

Like I explained in my first comment. This work the first time you create a client. But not when you need to edit the configuration. Just follow the step I describe in my comment and you will see the edited field are not propagated to the zos_protocol client

Comment (abom) [reply Jumpscale/clients/zero_os/ZeroOSClient.py:7]: @arahmanhamdy and @serboctor told me the data can be updated, there's something that supports this.

Will try to fix this.

Comment (abom) [reply Jumpscale/clients/zero_os/ZeroOSClient.py:7]: Should be resolved by https://github.com/threefoldtech/jumpscaleX/pull/294/commits/15714b393fd54050cfece2780266e9fa98bed40a

**Review by rkhamis — APPROVED** [review]

**Pull request merged**